File: Application/src/main/java/com/shareyourproxy/api/domain/factory/RealmChannelFactory.java
package com.shareyourproxy.api.domain.factory;

import com.shareyourproxy.api.domain.model.Channel;
import com.shareyourproxy.api.domain.realm.RealmChannel;
import com.shareyourproxy.api.domain.realm.RealmChannelType;

import java.util.HashMap;
import java.util.Map;

import io.realm.RealmList;

/**
 * Factory for creating {@link RealmChannel}s.
 */
public class RealmChannelFactory {

    public static RealmList<RealmChannel> getRealmChannels(HashMap<String, Channel> channels) {
        if (channels == null) {
            return null;
        }
        RealmList<RealmChannel> realmChannelArray;
        realmChannelArray = new RealmList<>();
        for (Map.Entry<String, Channel> channelEntry : channels.entrySet()) {
            Channel channel = channelEntry.getValue();
            RealmChannel realmChannel = new RealmChannel();
            RealmChannelType realmChannelType = new RealmChannelType();
            //construct the newChannel section

            //construct the newChannel type
            realmChannelType.setWeight(channel.channelType().getWeight());
            realmChannelType.setLabel(channel.channelType().getLabel());
            realmChannelType.setResId(channel.channelType().getResId());

            //construct the newChannel
            realmChannel.setId(channel.id().value());
            realmChannel.setLabel(channel.label());
            realmChannel.setChannelType(realmChannelType);
            realmChannel.setActionAddress(channel.actionAddress());

            //add to array
            realmChannelArray.add(realmChannel);
        }
        return realmChannelArray;
    }

}

File: Application/src/main/kotlin/com/shareyourproxy/api/domain/factory/RealmChannelFactory.kt
package com.shareyourproxy.api.domain.factory

import com.shareyourproxy.api.domain.model.Channel
import com.shareyourproxy.api.domain.realm.RealmChannel
import com.shareyourproxy.api.domain.realm.RealmChannelType
import io.realm.RealmList
import java.util.*

/**
 * Factory for creating [RealmChannel]s.
 */
internal object RealmChannelFactory {
    fun getRealmChannels(channels: HashMap<String, Channel>): RealmList<RealmChannel> {
        val realmChannelArray: RealmList<RealmChannel>
        realmChannelArray = RealmList<RealmChannel>()
        for (channelEntry in channels.entries) {
            val channel = channelEntry.value
            val realmChannel = RealmChannel()
            val realmChannelType = RealmChannelType()
            //construct the newChannel section

            //construct the newChannel type
            realmChannelType.weight = channel.channelType.weight
            realmChannelType.label = channel.channelType.label

            //construct the newChannel
            realmChannel.id = channel.id
            realmChannel.label = channel.label
            realmChannel.channelType = realmChannelType
            realmChannel.actionAddress = channel.actionAddress
            realmChannel.isPublic = channel.isPublic

            //add to array
            realmChannelArray.add(realmChannel)
        }
        return realmChannelArray
    }
}
